sheepdog/utils/__init__.py

```diff
@@ -164,8 +164,10 @@ def check_and_call(program, project, *args, **kwargs):
             )
             if not project_node:
                 raise NotFoundError('Project {} not found'.format(project))
-            flask.g.dbgap_accession_numbers = "{},{}".format(program_node.dbgap_accession_number,
-                                                             project_node.dbgap_accession_number)
+            flask.g.dbgap_accession_numbers = program_node.dbgap_accession_number or ""
+            if project_node.dbgap_accession_number is not None:
+                flask.g.dbgap_accession_numbers = "{},{}".format(flask.g.dbgap_accession_numbers,
+                                                                 project_node.dbgap_accession_number).strip(',')
```

Comment:
this code logic is really not well written to be readable...

Reply:
The logic is:
- if program has dbgap_access_number assign it to the aggregation number, otherwise skip
- then if the project has dbgap_access_number append it to the aggregation one. with `,`. Then strip it.
- else if project doesn't have one, don't need to do anything.
I think it is a correct logic

Reply:
not saying it's wrong, but it's not very readable